Dockerfile

```diff
@@ -12,34 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21.3@sha256:02d7116222536a5cf0fcf631f90b507758b669648e0f20186d2dc94a9b419a9b AS builder
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.21@sha256:98a0ff138c536eee98704d6909699ad5d0725a20573e2c510a60ef462b45cce0 AS builder
 ENV APP_ROOT=/opt/app-root
 ENV GOPATH=$APP_ROOT
 
 WORKDIR $APP_ROOT/src/
-ADD go.mod go.sum $APP_ROOT/src/
-RUN go mod download
 
-# Add source code
-ADD ./cmd/ $APP_ROOT/src/cmd/
-ADD ./pkg/ $APP_ROOT/src/pkg/
+COPY . .
```

Comment:
Personally I recommend you swap out the `COPY` command for `ADD`. If you keep the strategy they started with, the image should be significantly smaller as you dont have to load every file from the directory in. This should work if you create the directories as metioned above.
```suggestion
ADD go.mod go.sum $APP_ROOT/src/
RUN go mod download

# Add source code
ADD ./cmd/ $APP_ROOT/src/cmd/
ADD ./pkg/ $APP_ROOT/src/pkg/
```